### PR TITLE
MM-25762 - Moved password settings to client limited configuration

### PR DESCRIPTION
**Pull request by marianunez**
#### Summary
- Moved password settings to be passed down also in the client limited configuration when the user is no yet logged in so the webApp can properly indicate password requirements at account creation. 
- Removed setting defaults given that `PasswordSettings` is not a licensed feature that need to have a pre-set value.

#### Ticket Link
[MM-25762](https://mattermost.atlassian.net/browse/MM-25762)

### CODE REVIEW

File: config/client.go [hunk 1]
```diff
@@ -114,11 +114,6 @@ func GenerateClientConfig(c *model.Config, diagnosticID string, license *model.L
 	props["SamlPositionAttributeSet"] = "false"
 	props["EnableCluster"] = "false"
 	props["EnableMetrics"] = "false"
-	props["PasswordMinimumLength"] = "0"
-	props["PasswordRequireLowercase"] = "false"
-	props["PasswordRequireUppercase"] = "false"
-	props["PasswordRequireNumber"] = "false"
-	props["PasswordRequireSymbol"] = "false"
```
Comment (jupenur) [inline config/client.go:121]: Why was this here in the first place?

Comment (marianunez) [reply config/client.go:121]: I'm not sure, it sets defaults values for features that are EE. Maybe password settings used to be one of them in the past? 🤷‍♀️
